src/core/structure/sheet.ts

```diff
@@ -499,19 +628,28 @@ export default class Sheet {
     }
 
     if (!formatterChanged) return;
-
     // Apply new formatter to all cells in this group.
     for (const [opposingKey] of group.cellIndex) {
       const cell = this.cellData.get(group.cellIndex.get(opposingKey)!)!;
       if (group instanceof Column) {
         this.applyCellFormatter(cell, group.key, opposingKey as RowKey);
-        continue;
+        this.emitSetCellEvent(
+          this.getColumnIndex(group.key as ColumnKey)!,
+          this.getRowIndex(opposingKey as RowKey)!,
+          cell,
+        );
+      } else {
+        this.applyCellFormatter(
+          cell,
+          opposingKey as ColumnKey,
+          group.key as RowKey,
+        );
+        this.emitSetCellEvent(
+          this.getColumnIndex(opposingKey as ColumnKey)!,
+          this.getRowIndex(group.key as RowKey)!,
+          cell,
+        );
       }
-      this.applyCellFormatter(
-        cell,
-        opposingKey as ColumnKey,
-        group.key as RowKey,
-      );
     }
```

Comment:
```ts
    for (const [opposingKey] of group.cellIndex) {
      const cell = this.cellData.get(group.cellIndex.get(opposingKey)!)!;
      const colKey = (
        group instanceof Column ? group.key : opposingKey
      ) as ColumnKey;
      const rowKey = (group instanceof Row ? group.key : opposingKey) as RowKey;
      this.applyCellFormatter(cell, colKey, rowKey);
      this.emitSetCellEvent(
        this.getColumnIndex(colKey)!,
        this.getRowIndex(rowKey)!,
        cell,
      );
    }
```

Having the column/row logic handled separately defeats the purpose of having a generic method. Though this is not that pretty either :P